### ISSUE TITLE
Accept impl serde::Serialize

### ISSUE THREAD

**Issue by mike-lloyd03** [ctron/colored_json#6]
Is it possible to implement colorizing anything that implements `serde::Serialize` so I don't have to convert a struct to a `serde_json::Value` first? This step isn't terribly overburdening, but it alphabetizes the keys of my struct and I'd like to keep them in the same order.

Example:

```rust
#[derive(serde::Serialize)]
struct Item {
    name: String,
    desc: String,
}

let item = Item {
    name: "Bob",
    desc: "Is a man",
};
```
Using `serde`'s pretty print method:
```rust
let json_str = serde_json::to_string_pretty(&item)?;
println!("{json_str}");
```
Produces:
```json
{
    "name": "Bob",
    "desc": "Is a man"
}
```

But casting to `Value` and using `colored_json`:
```rust
let json_value = serde_json::json!(item);
let json_str = to_colored_json_auto(&json_value)?;
println!("{json_str}");
```

Produces:
```json
{
    "desc": "Is a man",
    "name": "Bob"
}
```

**Comment by ctron** [comment]
I think that would be a nice addition. Could you come up with a PR for this?

**Comment by ctron** [comment]
Ok, actually that was rather simple. Should be released shortly as `3.3.0`.

**Comment by ctron** [comment]
Released `3.3.0`

**Comment by mike-lloyd03** [comment]
Awesome! Thanks so much!
